MetasysServices/Models/MetasysObject.cs

```diff
@@ -65,8 +65,8 @@ public class MetasysObject : Utils.ObjectUtil
         /// <summary>
         /// The number of direct children objects.
         /// </summary>
-        /// <value>The number of children or -1 if there is no children data.</value>
-        public int ChildrenCount { set; get; }
+        /// <value>The number of children.</value>
+        public int ChildrenCount => Children.Count();
```

Comment:
It was odd that this had a setter. Also I didn't find any code that would ever set this to -1 (and I can't think why we'd want to do that.) I considered deleting this property but that's a breaking change.

Reply:
I was unclear about `ChildrenCount` too @michaelgwelch , so I have avoided using it myself. My only thought as to why it might have a setter is that perhaps when many levels of depth are retrieved, then it can include "grandchildren", etc.

Reply:
That may have been the idea @PeteHazelberg but nothing was ever setting it. So the comment was wrong and if we want to add a setter at some point it should be private.